internal/services/costmanagement/anomaly_alert_resource.go

```diff
@@ -39,6 +40,13 @@ func (AnomalyAlertResource) Arguments() map[string]*pluginsdk.Schema {
 			ValidateFunc: validation.StringIsNotEmpty,
 		},
 
+		"subscription_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
```

Comment:
this makes this a breaking change, also should this simply be scope allowing scopes other then a subscription to be used?

Reply:
Thanks for the feedback @katbyte, I've pushed an update that attempts to make the subscription id optional. I'm not 100% certain if this approach is the right one so I'm open to any feedback you have. I have a slight concern that the Read method may not be correct as when running the newly added complete test I have had to add the subscription id as an ignored field on the ImportStep as it was not being returned during that step of the test.

As for the question of whether this should be renamed to scope instead, I don't think so as it is not currently possible to create these alerts for any scope other than subscription. In addition, this naming is consistent with the other existing cost management resources which all have their scope attributes named in a specific rather than generic way.